clang/lib/Driver/Driver.cpp

```diff
@@ -6753,6 +6753,19 @@ const char *Driver::GetNamedOutputPath(Compilation &C, const JobAction &JA,
         &JA);
   }
 
+  // Redirect output for the generated source + integration footer.
+  if (isa<AppendFooterJobAction>(JA) && !isSaveTempsEnabled()) {
```

Comment:
Do I get it correctly that trying to retain the custom output directory for `-save-temps` would break the temporary file tracking infrastructure unless additional work is done?

Reply:
Using `-save-temps` causes the generated source+footer file to be created in the expected `$CWD` with the rest of the files.  I thought it made more sense just to retain that behavior instead of using the specified dir.  Do you have any preference here?

Reply:
AFAIMC, the two behavior types with `-save-temps` are equally fine; however, would it make sense to add a warning in case the current approach is retained?